src/main/java/net/earthcomputer/clientcommands/command/CTickCommand.java

```diff
@@ -0,0 +1,210 @@
+    private static class TickMeasuringTask extends SimpleTask {
+
+        private static final int PERIOD = 100;
+
+        private static final Minecraft minecraft = Minecraft.getInstance();
+
+        private int tickCount = 0;
+        private long totalTickTime = 0;
+        private long startTickTime;
+        private boolean hadFirstTick = false;
+        private long firstTickStart;
+        private long lastTickStart;
+
+        private final boolean tps;
+        private final boolean forceInaccurate;
+
+        public TickMeasuringTask(boolean tps, boolean forceInaccurate) {
+            this.tps = tps;
+            this.forceInaccurate = forceInaccurate;
+        }
+
+        public void incrTickCount(int count) {
+            if (!hadFirstTick) {
+                firstTickStart = System.nanoTime();
+                hadFirstTick = true;
+            } else {
+                tickCount += count;
+            }
+        }
+
+        public void startTick() {
+            startTickTime = System.nanoTime();
+            if (!hadFirstTick) {
+                firstTickStart = startTickTime;
+                hadFirstTick = true;
+            }
+        }
+
+        public void endTick() {
+            if (hadFirstTick) {
+                totalTickTime += System.nanoTime() - startTickTime;
+                tickCount++;
+            }
+        }
+
+        @Override
+        protected void onTick() {
+            if (tickCount >= PERIOD) {
+                lastTickStart = System.nanoTime();
+                _break();
+            }
+        }
+
+        @Override
+        public void initialize() {
+            minecraft.player.displayClientMessage(Component.translatable("commands.ctick.measuring"), false);
+        }
+
+        @Override
+        public void onCompleted() {
+            if (tps) {
+                long totalTime = lastTickStart - firstTickStart;
+                double tps = 1_000_000_000D * tickCount / totalTime;
+                minecraft.player.displayClientMessage(Component.translatable("commands.ctick.tps", totalTime == 0 ? Component.translatable("commands.ctick.tps.immeasurable") : DEC_FMT.format(tps)), false);
+            } else if (forceInaccurate) {
+                long totalTime = lastTickStart - firstTickStart;
+                double mspt = totalTime / (1_000_000D * tickCount);
+                minecraft.player.displayClientMessage(Component.translatable("commands.ctick.mspt", DEC_FMT.format(mspt)), false);
+                minecraft.player.displayClientMessage(Component.translatable("commands.ctick.mspt.inaccurate"), false);
+            } else {
+                double mspt = totalTickTime / (1_000_000D * tickCount);
+                minecraft.player.displayClientMessage(Component.translatable("commands.ctick.mspt", DEC_FMT.format(mspt)), false);
+            }
+        }
+
+        @Override
+        public boolean condition() {
+            return tickCount <= 1200;
```

Comment:
This wasn't a problem in 1.12 since the client tick rate was always expected to be 20 and hardly ever fell below 20, but now if the TPS is set to 1 then this could take a long time to complete. I would suggest using real time instead. Just be extra careful about there being zero ticks elapsed and dividing by zero.